### PR TITLE
init domain service

### PR DESCRIPTION
**Pull request by kaka89**

### CODE REVIEW

File: src/main/java/io/reactivesw/admin/authentication/domain/model/Admin.java [hunk 1]
```diff
@@ -77,6 +77,149 @@
   /**
    * List of roles.
    */
-  @ElementCollection
-  private List<String> roles;
+  @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/model/Admin.java]: List fetch should be lazy.

File: src/main/java/io/reactivesw/admin/authentication/domain/model/Role.java [hunk 2]
```diff
@@ -68,6 +70,6 @@
   /**
    * List of scopes.
    */
-  @ElementCollection
-  private List<String> scopes;
+  @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/model/Role.java]: List fetch should be lazy.

File: src/main/java/io/reactivesw/admin/authentication/domain/model/Scope.java [hunk 3]
```diff
@@ -69,11 +73,11 @@
...
   @ElementCollection
-  private List<String> permissions;
+  private List<Permission> permissions;
 
   /**
    * List of modules.
    */
-  @ElementCollection
-  private List<String> modules;
+  @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/model/Scope.java]: List fetch should be lazy.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/AdminService.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  private transient AdminRepository adminRepository;
+
+  /**
+   * Create admin with sample.
+   *
+   * @param admin Admin
+   * @return Admin
+   */
+  public Admin create(Admin admin) {
+    LOG.debug("Enter. admin: {}.", admin);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/AdminService.java:34]: should be Admin.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/AdminService.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    Admin savedAdmin = this.adminRepository.save(admin);
+
+    LOG.debug("Exit. admin: {}.", savedAdmin);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/AdminService.java]: should be Admin.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ModuleService.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  private transient ModuleRepository moduleRepository;
+
+  /**
+   * Create module with sample.
+   *
+   * @param module Module
+   * @return Module
+   */
+  public Module create(Module module) {
+    LOG.debug("Enter. module: {}.", module);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/ModuleService.java:34]: should be Module.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ModuleService.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    Module savedModule = this.moduleRepository.save(module);
+
+    LOG.debug("Exit. module: {}.", savedModule);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/ModuleService.java:38]: should be Module.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  private transient RoleRepository roleRepository;
+
+  /**
+   * Create module with sample.
+   *
+   * @param role Module
+   * @return Module
+   */
+  public Role create(Role role) {
+    LOG.debug("Enter. role: {}.", role);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java]: should be Role.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+   */
+  @Autowired
+  private transient RoleRepository roleRepository;
+
+  /**
+   * Create module with sample.
+   *
+   * @param role Module
+   * @return Module
+   */
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java]: this comment is for role or module?

File: src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  public Role create(Role role) {
+    LOG.debug("Enter. role: {}.", role);
+
+    Role savedRole = this.roleRepository.save(role);
+
+    LOG.debug("Exit. role: {}.", savedRole);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java:38]: should be Role.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+import io.reactivesw.admin.authentication.domain.model.Scope;
+import io.reactivesw.admin.authentication.infrastructure.repository.ScopeRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Role service, used to manage scopes.
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java]: this comment is for role or scope?

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+@Service
+public class ScopeService {
+
+  /**
+   * LOG.
+   */
+  private static final Logger LOG = LoggerFactory.getLogger(ScopeService.class);
+
+  /**
+   * Role repository.
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java]: comment should be scope.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+   */
+  @Autowired
+  private transient ScopeRepository scopeRepository;
+
+  /**
+   * Create module with sample.
+   *
+   * @param scope Module
+   * @return Module
+   */
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java]: this comment is for scope or module?

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+  public Scope create(Scope scope) {
+    LOG.debug("Enter. scope: {}.", scope);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java]: should be Scope.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+
+    Scope savedScope = this.scopeRepository.save(scope);
+
+    LOG.debug("Exit. scope: {}.", savedScope);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java:38]: should be Scope.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/enums/Permission.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+  /**
+   * permission for update a resource.
+   */
+  UPDATE,
+
+  /**
+   * permission for delete a resource.
+   */
+  DELETE
+
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/infrastructure/enums/Permission.java]: uppercase letter.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/AdminRepository.java [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Admin repository.
+ */
+@Repository
+public interface AdminRepository extends JpaRepository<Admin, String>,
+    CrudRepository<Admin, String> {
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/AdminRepository.java]: do not need use JpaRepository and CrudRepository both.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ModuleRepository.java [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Module repository.
+ */
+@Repository
+public interface ModuleRepository extends JpaRepository<Module, String>,
+    CrudRepository<Module, String> {
+}
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ModuleRepository.java]: do not need use JpaRepository and CrudRepository both.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/RoleRepository.java [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Role repository.
+ */
+@Repository
+public interface RoleRepository extends JpaRepository<Role, String>,
+    CrudRepository<Role, String> {
+}
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/RoleRepository.java]: do not need use JpaRepository and CrudRepository both.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ScopeRepository.java [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Scope repository.
+ */
+@Repository
+public interface ScopeRepository extends JpaRepository<Scope, String>,
+    CrudRepository<Scope, String> {
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ScopeRepository.java]: do not need use JpaRepository and CrudRepository both.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/util/JsonUtils.java [hunk 21]
```diff
@@ -126,7 +126,7 @@ public static boolean isJSONValid(String string) {
       try {
         OBJECT_MAPPER.readValue(value, clazz);
         result = true;
-      } catch (IOException e) {
+      } catch (IOException ex) {
         // do nothing
         LOG.debug("string is not valid json. string: {}", value);
```
Comment (davis999) [inline src/main/java/io/reactivesw/admin/authentication/infrastructure/util/JsonUtils.java]: uppercase letter.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java (hunk 13 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java (hunk 12 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java (hunk 11 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/domain/service/ScopeService.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java (hunk 9 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/domain/service/RoleService.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/domain/model/Scope.java (hunk 3 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/domain/model/Scope.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/domain/model/Role.java (hunk 2 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/domain/model/Role.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/domain/model/Admin.java (hunk 1 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/domain/model/Admin.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/enums/Permission.java (hunk 16 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/infrastructure/enums/Permission.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ModuleRepository.java (hunk 18 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ModuleRepository.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/RoleRepository.java (hunk 19 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/RoleRepository.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ScopeRepository.java (hunk 20 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/infrastructure/repository/ScopeRepository.java]: done.

File: src/main/java/io/reactivesw/admin/authentication/infrastructure/util/JsonUtils.java (hunk 21 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/admin/authentication/infrastructure/util/JsonUtils.java]: done.

**Pull request merged**